### PR TITLE
Bytes size check for tx hash

### PR DESCRIPTION
**Pull request by robin-thomas**
### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libMessage/Messenger.cpp [hunk 1]
```diff
@@ -113,6 +113,20 @@ void NumberToArray(const T& number, vector<unsigned char>& dst,
   Serializable::SetNumber<T>(dst, offset, number, S);
 }
 
+template <class K, class V>
+bool CopyWithSizeCheck(const K& arr, V& result) {
```
Comment (ShengguangXiao) [inline src/libMessage/Messenger.cpp]: Can add some unit test case for this function?

File: src/libMessage/Messenger.cpp [hunk 2]
```diff
@@ -113,6 +113,20 @@ void NumberToArray(const T& number, vector<unsigned char>& dst,
...
+  LOG_MARKER();
+
+  // Fixed length copying.
+  if (arr.size() != result.max_size()) {
```
Comment (ShengguangXiao) [inline src/libMessage/Messenger.cpp]: Why it is not if (arr.size() <= result.max_size()) ?

Comment (ansnunez) [reply src/libMessage/Messenger.cpp]: This function will be used for all our fixed-length types (the hashes).  As such, we shouldn't really expect to receive an array that is not exactly the size expected.  So it's better to have the strict `!=` check here.

File: src/libMessage/Messenger.cpp (hunk 1 above)
Comment (ansnunez) [reply src/libMessage/Messenger.cpp]: It's a bit tricky I think to add unit test case for this as it is a local function that does not need to be exposed outside of Messenger.cpp.  For now I think the easiest way to test is to run testnet and trigger the call to this function in all the other functions (I'll do this for @robin-thomas)

File: src/libMessage/Messenger.cpp (hunk 2 above)
Comment (ShengguangXiao) [reply src/libMessage/Messenger.cpp]: I see

File: src/libMessage/Messenger.cpp (hunk 1 above)
Comment (ShengguangXiao) [reply src/libMessage/Messenger.cpp]: We can put this function in Messenger.h and then can add unit test for it in unit test cpp files.

Comment (ansnunez) [reply src/libMessage/Messenger.cpp]: Updated as suggested

**Pull request merged**